lib/cn/csched_sp3.c

```diff
@@ -1747,61 +1581,34 @@ sp3_submit(struct sp3 *sp, struct cn_compaction_work *w, uint qnum)
-                 sp->samp_reduce ? " samp_reduce" : "");
-    }
+    if (debug_sched(sp) || (w->cw_debug & CW_DEBUG_START)) {
 
-    if (HSE_LIKELY(!w->cw_debug))
-        return;
-
-    if (cn_node_isroot(w->cw_node) && !(w->cw_debug & CW_DEBUG_ROOT))
-        return;
-    else if (cn_node_isleaf(w->cw_node) && !(w->cw_debug & CW_DEBUG_LEAF))
-        return;
-    else if (!(w->cw_debug & CW_DEBUG_INTERNAL))
-        return;
+        const struct cn_node_stats *ns = &w->cw_ns;
+        ulong hll_pct = cn_ns_keys(ns) ? ((100 * ns->ns_keys_uniq) / cn_ns_keys(ns)) : 0;
+        uint busycnt = atomic_read(&w->cw_node->tn_busycnt) >> 16;
 
-    slog_info(
-        HSE_SLOG_START("cn_comp_start"),
-        HSE_SLOG_FIELD("job", "%u", w->cw_job.sj_id),
-        HSE_SLOG_FIELD("comp", "%s", cn_action2str(w->cw_action)),
-        HSE_SLOG_FIELD("rule", "%s", cn_comp_rule2str(w->cw_comp_rule)),
-        HSE_SLOG_FIELD("cnid", "%lu", w->cw_tree->cnid),
-        HSE_SLOG_FIELD("lvl", "%u", w->cw_node->tn_loc.node_level),
-        HSE_SLOG_FIELD("off", "%u", w->cw_node->tn_loc.node_offset),
-        HSE_SLOG_FIELD("leaf", "%u", (uint)cn_node_isleaf(w->cw_node)),
-        HSE_SLOG_FIELD("qnum", "%u", w->cw_qnum),
-        HSE_SLOG_FIELD("c_nh", "%u", w->cw_nh),
-        HSE_SLOG_FIELD("c_nk", "%u", w->cw_nk),
-        HSE_SLOG_FIELD("c_nv", "%u", w->cw_nv),
-        HSE_SLOG_FIELD("c_kvsets", "%u", w->cw_kvset_cnt),
-        HSE_SLOG_FIELD("nd_kvsets", "%lu", (ulong)cn_ns_kvsets(&w->cw_ns)),
-        HSE_SLOG_FIELD("nd_cap%%", "%lu", (ulong)w->cw_ns.ns_pcap),
-        HSE_SLOG_FIELD("nd_keys", "%lu", (ulong)cn_ns_keys(&w->cw_ns)),
-        HSE_SLOG_FIELD(
-            "nd_hll%%",
-            "%lu",
-            (ulong)(
-                cn_ns_keys(&w->cw_ns) == 0
-                ? 0
-                : ((100 * w->cw_ns.ns_keys_uniq) / cn_ns_keys(&w->cw_ns)))),
-        HSE_SLOG_FIELD("rdsz_b", "%ld", (long)w->cw_est.cwe_read_sz),
-        HSE_SLOG_FIELD("wrsz_b", "%ld", (long)w->cw_est.cwe_write_sz),
-        HSE_SLOG_FIELD("i_alen_b", "%ld", (long)w->cw_est.cwe_samp.i_alen),
-        HSE_SLOG_FIELD("l_alen_b", "%ld", (long)w->cw_est.cwe_samp.l_alen),
-        HSE_SLOG_END);
+        slog_info(
+            HSE_SLOG_START("cn_comp_start"),
+            HSE_SLOG_FIELD("job", "%u", w->cw_job.sj_id),
+            HSE_SLOG_FIELD("jcnt", "%u", spt->spt_job_cnt),
+            HSE_SLOG_FIELD("bcnt", "%u", busycnt),
+            HSE_SLOG_FIELD("qnum", "%u", w->cw_qnum),
+            HSE_SLOG_FIELD("reduce", "%d", sp->samp_reduce),
+            HSE_SLOG_FIELD("cnid", "%lu", w->cw_tree->cnid),
+            HSE_SLOG_FIELD("comp", "%s", cn_action2str(w->cw_action)),
+            HSE_SLOG_FIELD("rule", "%s", cn_comp_rule2str(w->cw_comp_rule)),
+            HSE_SLOG_FIELD("lvl", "%u", w->cw_node->tn_loc.node_level),
+            HSE_SLOG_FIELD("off", "%u", w->cw_node->tn_loc.node_offset),
+            HSE_SLOG_FIELD("c_nk", "%u", w->cw_nk),
```

Comment:
if `c_nh` isn't useful, can it be removed from the struct as well?

Reply:
Yeah. Prob remove.  Also, I noticed the alen is 32M, are they being allocated with preallocate=false?  Seems like they should be.

Reply:
Will review an do in a different PR.  Not a fan of overloading PRs with unrelated work.